corecfg/proxy.go

```diff
@@ -0,0 +1,74 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package corecfg
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+var proxyConfigKeys = map[string]bool{
+	"http":  true,
+	"https": true,
+	"ftp":   true,
+}
+
+func etcEnvironment() string {
+	return filepath.Join(dirs.GlobalRootDir, "/etc/environment")
```

Comment:
well, i would expect a snap "core" config to operate on the whole snap environment (not on the whole system) so that snap services and apps use that setting as well as snapd does (read: snap-confine should make use of $CORE_SNAP/etc/environment when executing snaps')...

i agree it should not touch the hosts /etc/environment on classic though...